lib/kernel/src/inet.erl

```diff
@@ -1452,11 +1452,14 @@ fdopen(Fd, Addr, Port, Opts, Protocol, Family, Type, Module) ->
 %%  socket stat
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 
+-spec i() -> ok.
 i() -> i(tcp), i(udp), i(sctp).
 
+-spec i(socket_protocol()) -> ok.
```

Comment:
The type `socket_protocol()` is not documented so the link in the documentation that is created from this -spec gets broken.  Our link checker detected that.

The right(tm) way to fix this I guess would be to -export_type (only exported types are documented) the type `socket_protocol()` and document it in `lib/kernel/doc/src/inet.xml` (maybe right after `address_family ()`), but then it would be strange that the last type in that section `socket_type()` is undocumented, so -export_type and document that too.

Either that or in a coward and maintainability-hostile dirty fix replace `socket_protocol()` with `'tcp' | 'udp' | 'sctp'`.

Reply:
Thank you for your very comprehensible suggestion, I'm glad that there's a link checker to prevent broken links. How can I run it locally or see it output if it's part of the CI?

I exported `socket_protocol()` and added a datatype tag for it in `inet.xml`. Please let me know if there's anything else I should change.

Reply:
@RaimoNiskanen I'm interested in your link checker too (for erldocs.com).

Reply:
Sorry about the delay. I unfortunately have had more (o)pressing matters to attend to...

Your branch had fallen out of the daily builds when you pushed your corrections.  It is supposed to work that way, but I failed to re-instate the branch when that happened.  Now it is back in.

Regarding the link checker, @Zorbash and @fenollp...  I managed to find it.  It is unfortunately a very old perl4 script living in our daily build lab, written by a guy that left us more than 10 years ago, with unclear copyright, that use regular expressions to parse the HTML and is therefore quite brittle.  So it is runs in our CI framework, but is not part of our release.  It does it's duty well, but is not really fit to be shown to the public. :-/